src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -0,0 +1,313 @@
+
+import ai.rapids.cudf.ColumnVector;
+import ai.rapids.cudf.DType;
+import ai.rapids.cudf.HostColumnVector;
+import ai.rapids.cudf.Table;
+
+public class GpuTimeZoneDB {
+
+  public static final int TIMEOUT_SECS = 300;
+
+
+  // For the timezone database, we store the transitions in a ColumnVector that is a list of 
+  // structs. The type of this column vector is:
+  //   LIST<STRUCT<utcInstant: int64, localInstant: int64, offset: int32>>
+  private CompletableFuture<Map<String, Integer>> zoneIdToTableFuture;
+  private CompletableFuture<HostColumnVector> fixedTransitionsFuture;
+
+  private boolean closed = false;
+
+  GpuTimeZoneDB() {
+    zoneIdToTableFuture = new CompletableFuture<>();
+    fixedTransitionsFuture = new CompletableFuture<>();
+  }
+  
+  private static GpuTimeZoneDB instance = new GpuTimeZoneDB();
+  static GpuTimeZoneDB getInstance() {
+    return instance;
+  }
+  
+  /**
+   * Start to cache the database. This should be called on startup of an executor. It should start
+   * to cache the data on the CPU in a background thread. It should return immediately and allow the
+   * other APIs to be called. Depending on what we want to * do we can have the other APIs block
+   * until this is done caching, or we can have private APIs that would let us load and use specific
+   * parts of the database. I prefer the former solution at least until we see a performance hit
+   * where we are waiting on the database to finish loading.
+   */
+  public static void cacheDatabase() {
+    if (!instance.isLoaded()) {
+      Executor executor = Executors.newSingleThreadExecutor(
+        new ThreadFactory() {
+          private ThreadFactory defaultFactory = Executors.defaultThreadFactory();
+
+          @Override
+          public Thread newThread(Runnable r) {
+            Thread thread = defaultFactory.newThread(r);
+            thread.setName("gpu-timezone-database-0");
+            thread.setDaemon(true);
+            return thread;
+          }
+        });
+      instance.loadData(executor);
+    }
+  }
+
+
+  public static void shutdown() {
+    if (instance.isLoaded()) {
+      instance.close();
+    }
+  }
+
+  public static ColumnVector fromTimestampToUtcTimestamp(ColumnVector input, ZoneId currentTimeZone) {
+    // TODO: Remove this check when all timezones are supported
+    // (See https://github.com/NVIDIA/spark-rapids/issues/6840)
+    if (!isSupportedTimeZone(currentTimeZone)) {
+      throw new IllegalArgumentException(String.format("Unsupported timezone: %s",
+          currentTimeZone.toString()));
+    }
+    if (!instance.isLoaded()) {
+      cacheDatabase(); // lazy load the database
+    }
+    Integer tzIndex = instance.getZoneIDMap().get(currentTimeZone.normalized().toString());
+    Table transitions = instance.getTransitions();
+    ColumnVector result = new ColumnVector(convertTimestampColumnToUTC(input.getNativeView(),
+        transitions.getNativeView(), tzIndex));
+    transitions.close();
```

Comment:
I feel this is very expensive since we upload data to GPU and create a new transition table every time we call this function. Can we cache the transition table inside `instant`?

Reply:
Yeah. I plan to update this in a future PR since right now we just need to see if we are computing the right thing on the GPU. I think it's a still an open question as to how to cache it and what makes sense. I will file a follow up issue on optimizing this.

Reply:
Also, right now the functionality will be hidden behind a configuration flag, so there is time to optimize before fully exposing.

Reply:
Filed this https://github.com/NVIDIA/spark-rapids-jni/issues/1588